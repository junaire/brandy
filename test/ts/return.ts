var x = 1n;
var y = 2n;
var v = add2(x, y);
console.log(v);

function add2(x: bigint, y: bigint): bigint {
    return x + y;
}
