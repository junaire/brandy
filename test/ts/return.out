@main {
  v0: int = const 1;
  x: int = id v0;
  v1: int = const 2;
  y: int = id v1;
  v2: int = id x;
  v3: int = id y;
  v: int = call @add2 v2 v3;
  v: int = id v;
  v4: int = id v;
  print v4;
  v5: int = const 0;
}
@add2(x: int, y: int): int {
  v0: int = id x;
  v1: int = id y;
  v2: int = add v0 v1;
  ret v2;
}
